lib/rewards/server/plugin.js

```diff
@@ -0,0 +1,38 @@
+/**
+ *
+ * Reldens - Rewards Server Plugin
+ *
+ */
+const {PluginInterface} = require("../../features/plugin-interface");
+const {sc} = require("@reldens/utils");
+const {ObjectSubscriber} = require("./subscribers/object-subscriber");
+
+class RewardsPlugin extends PluginInterface
+{
+
+    setup(props)
+    {
+        this.events = sc.get(props, 'events', false);
+        if (!this.events) {
+            console.log('EventsManager undefined in RewardsPlugin.');
+        }
+
+        this.events.on('reldens.afterRunAdditionalRespawnSetup', async (eventArg) => {
+            await ObjectSubscriber.enrichWithRewards(eventArg.objInstance);
+        });
+
+        this.events.on('reldens.battleEnded', async (eventArg) => {
+            const {playerSchema, pve} = eventArg;
+            if (!pve.targetObject?.rewards) {
+                return false;
+            }
+            pve.targetObject.rewards.forEach(async (rewardModel) => {
```

Comment:
don't use .forEach, use `for of`, like:
```
for(let reward of pve.targetObject.rewards){
    /// your code.
}
```

Reply:
should we use const or let for the for of variable?

Reply:
as general rule I'm using `const` only for the imports, the difference with `let` is that `let` is context limited, for example:
```
let myvar = 1;
if(something){
console.log(myvar); // 1
}
```
if(something){
    let myvar = 1;
}
console.log(myvar); // undefined
```
so basically I use let for everything inside the classes

Reply:
got it!